core/src/main/java/edu/wpi/grip/core/StopStartSource.java

```diff
@@ -0,0 +1,52 @@
+package edu.wpi.grip.core;
+
+
+import com.google.common.eventbus.EventBus;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * A Source that can be stopped and started multiple times.
+ */
+public abstract class StopStartSource extends Source {
+
+    /**
+     * Starts this source.
+     *
+     * @return The source object that created the camera
+     * @throws IOException If the source fails to be started
+     */
+    public <T extends Source> T start(EventBus eventBus) throws IOException {
```

Comment:
This probably shouldn't return itself.  Yeah, you can use the return value to shorten the total number of lines of code you have to write, but doing it in a simpler way (create the object, then call a method on it, etc) would be better.